pkg/scaling/executor/scale_scaledobjects.go

```diff
@@ -85,6 +82,43 @@ func (e *scaleExecutor) RequestScale(ctx context.Context, scaledObject *kedav1al
 		}
 	}
 
+	// Check if we are paused, and if we are then update the scale to the desired count.
+	pausedCount, err := GetPausedReplicaCount(scaledObject)
+	if err != nil {
+		if err := e.setReadyCondition(ctx, logger, scaledObject, metav1.ConditionFalse,
+			kedav1alpha1.ScaledObjectConditionReadySucccesReason, kedav1alpha1.ScaledObjectConditionReadySuccessMessage); err != nil {
+			logger.Error(err, "error setting ready condition")
+		}
+		logger.Error(err, "Error getting the paused replica count on the current ScaledObject.")
+		return
+	}
+	if pausedCount != nil {
+		_, err := e.updateScaleOnScaleTarget(ctx, scaledObject, currentScale, *pausedCount)
```

Comment:
I think we should do this only `if pausedCount != currentReplicas`, this way we will do all these operations on every iteration.

Reply:
I think that we should check that ScaledObject has been already paused in here as well. To avoid the corner case when user set the pausedCount to the currentReplicas, but we haven't yet paused the scaling. WDYT?

Reply:
Yeah good thinking, this corner case didn't occur to me.